doc/adr/0013-synchronous_peer_discovery.md

```diff
@@ -0,0 +1,63 @@
+# 13. Synchronous Peer Discovery
```

Comment:
IMO the name is contrary to this sentence:
> This way, the peer discovery process itself is unmodified.

I'd name it as delayed download or something like that. Apart from that, I'm good with the ADR

Reply:
I'll change this in a separate PR.

Reply:
Change made in PR https://github.com/Constellation-Labs/tessellation/pull/632